phpBB/phpbb/extension/base.php

```diff
@@ -137,6 +139,30 @@ protected function get_migration_file_list()
 
 		$migrations = $this->extension_finder->get_classes_from_files($migrations);
 
+		if ($validate_classes)
+		{
+			foreach ($migrations as $key => $migration)
+			{
+				// If the class exists and is a subclass of the
+				// \phpbb\db\migration\migration abstract class
+				// we skip it.
+
+				// Otherwise, i.e. if it doesn't exist or it is
+				// not an extend the abstract class, we unset it
+				if (class_exists($migration))
+				{
+					$reflector = new \ReflectionClass($migration);
+					if ($reflector->isSubclassOf('\phpbb\db\migration\migration'))
```

Comment:
The ticket specifies that devs should be able to "...keep migration helpers / abstract classes they may want to use in their migrations folder (like the way phpBB is able to do)."

If the only thing we check for here is instantiability then any "migration helpers" (i.e. classes that don't extend the migration base class) will be treated as migrations as well. Your use case is classes that "...still extend the migration class but are meant to be extended themselves"

My ticket currently covers both criteria. Or am I misunderstanding?